### PR TITLE
feat: bump velero v1.10.1

### PR DESCRIPTION
**Pull request by rajgitnutanix**
**What problem does this PR solve?**:
Bump to
 velero/velero-plugin-for-aws:v1.10.1
**Which issue(s) does this PR fix?**:

https://jira.nutanix.com/browse/NCN-102803

trivy image docker.io/velero/velero-plugin-for-aws:v1.10.1
2024-10-11T12:51:43+05:30	INFO	[db] Need to update DB
2024-10-11T12:51:43+05:30	INFO	[db] Downloading DB...	repository="ghcr.io/aquasecurity/trivy-db:2"
54.17 MiB / 54.17 MiB [--------------------------------------------------------------------------------------] 100.00% 3.07 MiB p/s 18s
2024-10-11T12:52:03+05:30	INFO	[vuln] Vulnerability scanning is enabled
2024-10-11T12:52:03+05:30	INFO	[secret] Secret scanning is enabled
2024-10-11T12:52:03+05:30	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-10-11T12:52:03+05:30	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.55/docs/scanner/secret#recommendation for faster secret detection
2024-10-11T12:52:15+05:30	INFO	Number of language-specific files	num=2
2024-10-11T12:52:15+05:30	INFO	[gobinary] Detecting vulnerabilities...
2024-10-11T12:52:15+05:30	WARN	Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.55/docs/scanner/vulnerability#severity-selection for details.

bin/cp-plugin (gobinary)

Total: 3 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 1, CRITICAL: 0)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬────────────────┬─────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version  │                            Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼────────────────┼─────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-34156 │ HIGH     │ fixed  │ 1.22.6            │ 1.22.7, 1.23.1 │ encoding/gob: golang: Calling Decoder.Decode on a message   │
│         │                │          │        │                   │                │ which contains deeply nested structures...                  │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34156                  │
│         ├────────────────┼──────────┤        │                   │                ├─────────────────────────────────────────────────────────────┤
│         │ CVE-2024-34155 │ MEDIUM   │        │                   │                │ go/parser: golang: Calling any of the Parse functions       │
│         │                │          │        │                   │                │ containing deeply nested literals...                        │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34155                  │
│         ├────────────────┤          │        │                   │                ├─────────────────────────────────────────────────────────────┤
│         │ CVE-2024-34158 │          │        │                   │                │ go/build/constraint: golang: Calling Parse on a "// +build" │
│         │                │          │        │                   │                │ build tag line with...                                      │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34158                  │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴────────────────┴─────────────────────────────────────────────────────────────┘

plugins/velero-plugin-for-aws (gobinary)

Total: 3 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 1, CRITICAL: 0)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬────────────────┬─────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version  │                            Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼────────────────┼─────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-34156 │ HIGH     │ fixed  │ 1.22.6            │ 1.22.7, 1.23.1 │ encoding/gob: golang: Calling Decoder.Decode on a message   │
│         │                │          │        │                   │                │ which contains deeply nested structures...                  │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34156                  │
│         ├────────────────┼──────────┤        │                   │                ├─────────────────────────────────────────────────────────────┤
│         │ CVE-2024-34155 │ MEDIUM   │        │                   │                │ go/parser: golang: Calling any of the Parse functions       │
│         │                │          │        │                   │                │ containing deeply nested literals...                        │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34155                  │
│         ├────────────────┤          │        │                   │                ├─────────────────────────────────────────────────────────────┤
│         │ CVE-2024-34158 │          │        │                   │                │ go/build/constraint: golang: Calling Parse on a "// +build" │
│         │                │          │        │                   │                │ build tag line with...                                      │
│         │                │          │        │                   │                │ https://avd.aquasec.com/nvd/cve-2024-34158                  │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴────────────────┴─────────────────────────────────────────────────────────────┘

### CODE REVIEW

**Review by mhrabovcin — CHANGES_REQUESTED** [review]
The relevant test from `kommander` repo is failing:

```
    logger.go:42: 06:51:10 | backup-restore/1-create-backup | starting test step 1-create-backup
    logger.go:42: 06:51:10 | backup-restore/1-create-backup | Backup:kommander/velero-test-backup created
    logger.go:42: 06:56:11 | backup-restore/1-create-backup | test step failed 1-create-backup
```

**Comment by rajgitnutanix** [comment]
> The relevant test from `kommander` repo is failing:
> 
> ```
>     logger.go:42: 06:51:10 | backup-restore/1-create-backup | starting test step 1-create-backup
>     logger.go:42: 06:51:10 | backup-restore/1-create-backup | Backup:kommander/velero-test-backup created
>     logger.go:42: 06:56:11 | backup-restore/1-create-backup | test step failed 1-create-backup
> ```

Checking on the same as suggested.. Thank you..

**Comment by mhrabovcin** [comment]
@rajgitnutanix can we rebase this PR on `main` and re-run the `kommander` test with velero assertions?

**Comment by rajgitnutanix** [comment]
> @rajgitnutanix can we rebase this PR on `main` and re-run the `kommander` test with velero assertions?

sure will do that

File: services/velero/7.1.5/defaults/cm.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ data:
             s3Url: http://rook-ceph-rgw-dkp-object-store.${releaseNamespace}.svc:80/
             s3ForcePathStyle: true
             insecureSkipTLSVerify: "true"
+            checksumAlgorithm: "SHA256"
```
Comment (mhrabovcin) [inline services/velero/7.1.5/defaults/cm.yaml:29]: Can we run kommander tests with no checksum algo?
```suggestion
            checksumAlgorithm: ""
```

**Review by mhrabovcin — APPROVED** [review]

**Pull request merged**